src/components/HistoryPagination.vue

```diff
@@ -0,0 +1,92 @@
+<template>
+  <s-pagination
+    class="history-pagination"
+    layout="slot"
+    :current-page.sync="currentPage"
+    :page-size="pageAmount"
+    :total-text="totalText"
+  >
+    <span class="el-pagination__total">{{ totalText }}</span>
+    <s-button
+      type="link"
+      :tooltip="t('history.firstText')"
+      :disabled="isFirstPage"
+      @click="handlePaginationClick(PaginationButton.First)"
+    >
+      <s-icon name="chevrons-left-16" size="14" />
+    </s-button>
+    <s-button
+      type="link"
+      :tooltip="t('history.prevText')"
+      :disabled="isFirstPage"
+      @click="handlePaginationClick(PaginationButton.Prev)"
+    >
+      <s-icon name="chevron-left-16" size="14" />
+    </s-button>
+    <s-button
+      type="link"
+      :tooltip="t('history.nextText')"
+      :disabled="isLastPage"
+      @click="handlePaginationClick(PaginationButton.Next)"
+    >
+      <s-icon name="chevron-right-16" size="14" />
+    </s-button>
+    <s-button
+      type="link"
+      :tooltip="t('history.lastText')"
+      :disabled="isLastPage"
+      @click="handlePaginationClick(PaginationButton.Last)"
+    >
+      <s-icon name="chevrons-right-16" size="14" />
+    </s-button>
+  </s-pagination>
+</template>
+
+<script lang="ts">
+import { Mixins, Component, Prop } from 'vue-property-decorator';
+
+import TranslationMixin from './mixins/TranslationMixin';
+import { PaginationButton } from '../consts';
+
+@Component
+export default class HistoryPagination extends Mixins(TranslationMixin) {
+  @Prop({ default: 1, type: Number }) readonly currentPage!: number;
+  @Prop({ default: 10, type: Number }) readonly pageAmount!: number;
+  @Prop({ default: 'Total', type: String }) readonly totalText!: string;
```

Comment:
maybe we can add total pages number here instead of is first/last page props and 2 getters here like

get isFirst[Last]Page() // checks currentPage

you don't need to calc it somewhere else I guess

Reply:
but m not sure, if you don't agree - we can have a short call